src/composition.rs

```diff
@@ -0,0 +1,649 @@
+            // If the statement was found, don't keep looking.
+            return false;
+        }
+
+        if let ast::walk::Node::ExprStmt(expr_statement) = node {
+            if let ast::Expression::PipeExpr(expr) =
+                &expr_statement.expression
+            {
+                if let ast::Expression::Identifier(identifier) =
+                    &expr.call.callee
+                {
+                    if identifier.name == "yield" {
+                        expr.call.arguments.iter().any(|argument| {
+                            if let ast::Expression::Object(object) = argument {
+                                object.properties.iter().any(|property| {
+                                    if let ast::PropertyKey::Identifier(key) = &property.key {
+                                        if key.name == "name" {
+                                            // Because we would have generated this statement, we'll always be able to
+                                            // assert the simplicity of the yield name, i.e. it won't be a deeply nested
+                                            // expression.
+                                            if let Some(ast::Expression::StringLit(literal)) = &property.value {
+                                                if literal.value == YIELD_IDENTIFIER {
+                                                    self.statement = Some(expr_statement.clone());
+                                                    return true;
+                                                }
+                                            }
+                                        }
+                                    }
+                                    false
+                                })
+                            } else {
+                                false
+                            }
+                        });
+                        return false;
+                    }
+                }
+            }
+        }
+        true
+    }
+}
+
+type CompositionResult = Result<(), ()>;
+
+/// Composition acts as the public entry point into the composition functionality.
+struct Composition {
+    file: ast::File,
+}
+
+impl ToString for Composition {
+    fn to_string(&self) -> String {
+        flux::formatter::convert_to_string(&self.file)
+            .expect("Unable to convert composition file to string.")
+    }
+}
+
+impl Composition {
+    #[allow(dead_code)]
+    fn new(file: ast::File) -> Self {
+        Self { file }
+    }
+
+    /// Initialize an ast::File for use in composition.
+    ///
+    /// This must be called before any other composition can be made, as it'll set up the
+    /// statement that will be managed by composition.
+    #[allow(dead_code)]
+    fn initialize(
+        &mut self,
+        bucket: &str,
+        measurement: Option<&str>,
+    ) -> CompositionResult {
+        let mut visitor =
+            CompositionStatementFinderVisitor::default();
```

Comment:
If I understand correctly, it looks like it's identifying existing compositions based on the the existence of a yield statement (`YIELD_IDENTIFIER`).

This is the reset feature. So when re-initializing on an existing composition, it erases/replaces the old composition.

I believe `|> yield()` itself requires a unique identifier. So could not the user:
1. create a composition.
2. in the raw editor window, manually type in to change the yield identifier.
   * note: this stops the composition sync
3. then re-initialize a new composition by using the schema builder again?

So we are supporting the use of composition to create multiple subqueries, in the same flux. But that's technically NOT multiple compositions. It's still 1 composition per window...but the user can rename the `|> yield()` of the previous composition --> then create a new composition.

Reply:
Yeah, if you know our secret sauce, you can do whatever you want. Our tests do just that.

Reply:
Cool! Then we can see if "multiple composition" can be a UI concept. Not an LSP concept. Thanks!